sunkit_image/enhance.py

```diff
@@ -101,51 +101,42 @@ def mgn(
             "behavior.",
             stacklevel=3,
         )
-
     if weights is None:
         weights = np.ones(len(sigma))
-
     # 1. Replace spurious negative pixels with zero
     if clip is True:
         data[data <= 0] = 1e-15  # Makes sure that all values are above zero
-
+    data = data.astype(np.float32) if data.dtype != np.float32 else data
```

Comment:
Why float32? Why not double precision for example?

Reply:
I just picked one at random. I should just promote it to that instead. That makes more sense.

Reply:
Trying that I still get:

```
numpy._core._exceptions._UFuncOutputCastingError: Cannot cast ufunc 'sqrt' output from dtype('float64') to dtype('int64') with casting rule 'same_kind'
```

I can probably make it unsafe?

Reply:
I came up with worse ideas.